server/event_handler.py

```diff
@@ -0,0 +1,145 @@
+                        self.room = self.manager._rooms[room_code]
+                    case "join":
+                        self.manager.join_room(self.client, room_code)
+                        self.room = self.manager._rooms[room_code]
+
+                        collaborators = [{"id": c.id.hex, "username": c.username} for c in self.room.clients]
+
+                        # Send a sync event to the client to update the code and
+                        # the collaborators' list
+                        response = EventResponse(
+                            type=EventType.SYNC,
+                            data=SyncData(code=self.room.code, collaborators=collaborators),
+                            status_code=StatusCode.SUCCESS,
+                        )
+                        await self.client.send(response)
+
+                        # Broadcast to other clients a connect event to update
+                        # the collaborators' list
+                        response = EventResponse(
+                            type=EventType.CONNECT,
+                            data=connect_data,
+                            status_code=StatusCode.SUCCESS,
+                        )
+                        await self.manager.broadcast(response, room_code, sender=self.client)
+            case EventType.DISCONNECT:
+                self.manager.disconnect(self.client, room_code)
+
+                collaborators = [{"id": c.id.hex, "username": c.username} for c in self.room.clients]
+
+                # Broadcast to other clients a sync event to update the
+                # collaborators' list
+                response = EventResponse(
+                    type=EventType.SYNC,
+                    data=SyncData(code=self.room.code, collaborators=collaborators),
+                    status_code=StatusCode.SUCCESS,
+                )
+                await self.manager.broadcast(response, room_code, sender=self.client)
+
+                WebSocketDisconnect.response = response  # type: ignore
+                raise WebSocketDisconnect
+            case EventType.MOVE:
+                move_data = cast(MoveData, event_data)
+                self.room.cursors[self.client.id] = move_data.position
+
+                # Broadcast to every client a move event to update the cursors'
+                # positions
+                response = EventResponse(type=EventType.MOVE, data=move_data, status_code=StatusCode.SUCCESS)
+                await self.manager.broadcast(response, room_code)
+            case EventType.REPLACE:
+                replace_data = cast(ReplaceData, event_data)
+                self.room.update_code(replace_data)
+
+                # Broadcast to every client a replace event to update the code
+                response = EventResponse(type=EventType.REPLACE, data=replace_data, status_code=StatusCode.SUCCESS)
+                await self.manager.broadcast(response, room_code)
```

Comment:
`await self.manager.broadcast(response, room_code, sender=self.client)`.

The client sending the request doesn't need to receive their own update, only other clients in the room.